### PR TITLE
Add pallet-storage-roots

### PR DESCRIPTION
**Pull request by tmpolaczyk**

### CODE REVIEW

File: pallets/relay-storage-roots/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		#[pallet::constant]
+		type MaxStorageRoots: Get<u32>;
+		/// Weight info
+		type WeightInfo: WeightInfo;
+	}
+
+	/// Ensures the mandatory inherent was included in the block
+	#[pallet::storage]
+	#[pallet::getter(fn inherent_included)]
+	pub(crate) type InherentIncluded<T: Config> = StorageValue<_, ()>;
```
Comment (girazoki) [inline pallets/relay-storage-roots/src/lib.rs:60]: you can remove this now

File: pallets/relay-storage-roots/src/mock.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+		.expect("Pallet balances storage can be assimilated");
+
+		let mut ext = sp_io::TestExternalities::new(t);
+		ext.execute_with(|| System::set_block_number(1));
+		ext
+	}
+}
+
+pub const ALICE: H160 = H160::repeat_byte(0xAA);
+//pub const BOB: H160 = H160::repeat_byte(0xBB);
```
Comment (girazoki) [inline pallets/relay-storage-roots/src/mock.rs:190]:
```suggestion
```

File: pallets/relay-storage-roots/src/benchmarks.rs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+			number: 1000,
+			state_root: H256::default(),
+		};
+
+		T::RelaychainStateProvider::set_current_relay_chain_state(relay_state.clone());
+	}: {
+		Pallet::<T>::set_relay_storage_root()
+	}
+	verify {
+		// verify randomness result
```
Comment (girazoki) [inline pallets/relay-storage-roots/src/benchmarks.rs:56]:
```suggestion
```

**Comment by tmpolaczyk** [comment]
This has been merged upstream https://github.com/Moonsong-Labs/moonkit/pull/18

**Pull request closed without merging**